### PR TITLE
#80 Creates `tracer pkg` with first trace "AssertPersistedGenesisBlock".

### PR DESCRIPTION
**Pull request by EnchanterIO**
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests

### CODE REVIEW

File: database/init.go [hunk 1]
```diff
@@ -12,9 +12,10 @@ import (
 	ethCore "github.com/ethereum/go-ethereum/core"
 	tmtLog "github.com/tendermint/tendermint/libs/log"
 	"github.com/lightstreams-network/lightchain/log"
+	"github.com/lightstreams-network/lightchain/tracer/dbtracy"
 )
 
-func Init(cfg Config, ntw setup.Network) error {
+func Init(cfg Config, ntw setup.Network, dbTracer dbtracy.Tracer) error {
```
[the remark this answers is not in the document]
Comment (EnchanterIO) [reply database/init.go]: Gabri, you are looking at old code where the requirement was to separate tracer from the codebase. Even github tells you it's "outdated" :D

screen shot 2019-02-19 at 12 04 48

File: tracer/dbtracy/assert_genesis.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+)
+
+func (t EthDBTracer) AssertPersistedGenesisBlock(genesis core.Genesis) {
+	t.logger.Infow("Tracing if ETH DB wrote a valid genesis block to disk...", "chaindata", t.chainDataDir)
+
+	chainDb, err := ethdb.NewLDBDatabase(t.chainDataDir, 0, 0)
```
[the remark this answers is not in the document]
Comment (EnchanterIO) [reply tracer/dbtracy/assert_genesis.go]: There IS absolutely the need of checking for an error always. You can't rely on your methods on a previous stack execution which may change over time dude. You code in a method scope not in a global state....

File: TRACER.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+## Enable tracing
+
+Tracing can be enabled via the `--trace` flag and the log file can be configured using the `--tracelog` flag.
+```
+lightchain init --datadir=/Users/enchanterio/.lightchain --standalone --trace
+```
+
+Output example:
+```
+lukas:lightchain enchanterio$ tail -f /var/folders/f9/11wh9x9j31d75nxgwv2qsj3r0000gn/T/tracer.log
```
[the remark this answers is not in the document]
Comment (EnchanterIO) [reply TRACER.md]: ok done

File: node/config.go [hunk 4]
```diff
@@ -18,3 +18,7 @@ func NewConfig(dataDir string, consensusCfg consensus.Config, dbCfg database.Con
 		dbCfg,
 	}
 }
+
+func (c Config) DbChainDir() string {
```
[the remark this answers is not in the document]
Comment (EnchanterIO) [reply node/config.go]: actually the method is not used was a refactoring experiment that I missed to remove. Method removed in last commit.

**Comment by ggarri** [comment]
After an offline discussion, it was agreed to accept these changes provisionally till a more scalable solution is achieved

**Pull request merged**